### PR TITLE
feat: multiple data files

### PR DESCRIPTION
**Pull request by jjfrench**
Adds the ability to create STAC records from CMR collections containing granules with multiple data files.

- Isn't able to satisfy every single collection for every single granule file, some collections will have straggler files e.g ([.rdr files](https://search.maap-project.org/search/granules?p=C1200000308-NASA_MAAP&pg%5B0%5D%5Bid%5D=uavsar_AfriSAR_v1_SLC*.rdr&q=AfriSAR_UAVSAR_Coreg_SLC&m=0.0703125!-0.0703125!2!1!0!0%2C2&tl=1664910719!4!!))
- Matches most cases where granules contain files in a {product_id}_{data_file}.{ext} pattern

### CODE REVIEW

File: lambdas/build-stac/tests/conftest.py
Comment (jjfrench) [reply lambdas/build-stac/tests/conftest.py:61]: Not sure where you're seeing external calls. Can you show me?

Comment (emileten) [reply lambdas/build-stac/tests/conftest.py]: I just understood that : 
- since it's a CMR event, we wouldn't be reading the file in itself anyways
- in addition, you're using the `patch` method to skip the `GranuleQuery` call.

**Review by emileten — APPROVED** [review]
LGTM if this works for your immediate purposes. We can always work on future improvements to cover more cases?

**Comment by jjfrench** [comment]
@emileten, I added that content-type commit we were discussing on Slack. Otherwise, if I could get @sbquinlan's seal of approval on this PR - we should be good to merge

**Review by sbquinlan — APPROVED** [review]
seal of approval

File: lambdas/cmr-query/handler.py
Comment (abarciauskas-bgse) [inline lambdas/cmr-query/handler.py:90]: Can you explain a bit more about what this code is doing and why it's needed? This should be a non-blocker, just asking for my own understanding.

**Comment by abarciauskas-bgse** [comment]
Thanks @jjfrench , looks good to me as well - I don't know if you ever responded to @emileten question since some of the comments are hidden but I also wanted to make sure my understanding of what is intended to happen is correct: The search for granules is the same, returning a list of `granules_to_insert` but since that list (`granules_to_insert`) is 1 to 1 CMR granules to list items, the `multi_asset_items` groups granules using the `data_file_regex` as a way to identify which files should be grouped together (that is, when multiple files match that data file regex they should be grouped together).

File: lambdas/cmr-query/handler.py
Comment (jjfrench) [reply lambdas/cmr-query/handler.py:90]: Yeah! Sorry if the comments weren't clear enough, but essentially we're grouping all the files with the same product_id under a `Dict[product_id, Dict[file_name, List[str]]]` and then adding those grouped files `List[str]` to the granule that we want to build the STAC record around (`data_file`) and adding it to a different `List[Dict]` for the `return_obj`

**Comment by jjfrench** [comment]
> Thanks @jjfrench , looks good to me as well - I don't know if you ever responded to @emileten question since some of the comments are hidden but I also wanted to make sure my understanding of what is intended to happen is correct: The search for granules is the same, returning a list of `granules_to_insert` but since that list (`granules_to_insert`) is 1 to 1 CMR granules to list items, the `multi_asset_items` groups granules using the `data_file_regex` as a way to identify which files should be grouped together (that is, when multiple files match that data file regex they should be grouped together). Is that understanding correct?

Yup! We still grab everything from CMR but now `multi_asset_items` is grouping the files together to add onto the `pystac.Item.assets` based off the `data_file_regex` (typically the product_id) and `data_file`, the file to build the STAC record around per product. So if each product had only one hdr file and that's what you want to build the STAC record around, `data_file` would/could just be "hdr"

**Comment by jjfrench** [comment]
I've tested this on a couple of collections in the TEST stack, ABoVE_UAVSAR_PALSAR is a good example where it created 30 STAC records with 12 item assets. The AfriSAR collections were working well but it seems like the documentation link is unreachable for the time being but that shouldn't be a blocker for this PR since the code is working as intended. 

[Recap]
"data_file_regex" is a regex str pattern, typically a product ID
"data_file" is the data file name from which to build the STAC record around
"asset_media_type" can now be a dictionary of values of media types
"limit" should be either all granules or a multiple of the grouped number of assets from the CMR query

**Comment by abarciauskas-bgse** [comment]
@jjfrench thanks for the update.

If the documentation link for the afrisar collections is unreachable, I don't think we should include it in the metadata. Can we skip those links to documentation? And perhaps we can track down who to reach out to for updated documentation links. Do you have an example documentation link / CMR collection link handy?

**Pull request merged**
